website/docs/d/virtual_machine.html.markdown

```diff
@@ -3,29 +3,50 @@ layout: "vsphere"
 page_title: "VMware vSphere: vsphere_virtual_machine"
 sidebar_current: "docs-vsphere-data-source-virtual-machine"
 description: |-
-  Provides a vSphere virtual machine data source. This can be used to get data from a virtual machine or template.
+  Provides a VMware vSphere virtual machine data source. This can be used to return data from a virtual machine or template.
 ---
 
 # vsphere\_virtual\_machine
 
 The `vsphere_virtual_machine` data source can be used to find the UUID of an
-existing virtual machine or template. Its most relevant purpose is for finding
-the UUID of a template to be used as the source for cloning into a new
+existing virtual machine or template. THe most common purpose is for finding
+the UUID of a template to be used as the source for cloning to a new
 [`vsphere_virtual_machine`][docs-virtual-machine-resource] resource. It also
 reads the guest ID so that can be supplied as well.
 
 [docs-virtual-machine-resource]: /docs/providers/vsphere/r/virtual_machine.html
 
 ## Example Usage
 
+In the following example, a virtual machine template is returned by its
+unique name within the `vsphere_datacenter`.
+
 ```hcl
 data "vsphere_datacenter" "datacenter" {
-  name = "dc1"
+  name = "dc-01"
 }
 
 data "vsphere_virtual_machine" "template" {
-  name          = "test-vm-template"
-  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
+  name          = "ubuntu-server-template"
+  datacenter_id = data.vsphere_datacenter.datacenter.id
```

Comment:
Great question!

The examples uses the correct format. 

For example:

```hcl
data "vsphere_datacenter" "datacenter" {
   name = "dc-01"
}

data "vsphere_virtual_machine" "template" {
   name          = "ubuntu-server-template"
   datacenter_id = data.vsphere_datacenter.datacenter.id
}
```

This passes the UUID found by the `vsphere_datacenter.datacenter` data source with the name "dc-01" to the `vsphere_virtual_machine.template` data source as the `datacenter_id` to find the correct virtual machine template within the vSphere datacenter object. 

If it were in double-quotes it would be read as the literal "data.vsphere_datacenter.datacenter.id" instead of the UUID from the data source.

However, `"${data.vsphere_datacenter.dc.id}"` interpolation would also work, but this [was deprecated](https://github.com/hashicorp/terraform/blob/v0.12/CHANGELOG.md) in Terraform 0.12.14, if I recall. correctly.  In fact, a `terrafrom fmt` will replace `"${data.vsphere_datacenter.dc.id}"` with `data.vsphere_datacenter.dc.id`.

One of the things I'd like to see (_and happy to help with_) is to update all examples with a consistent style and context that also removes the legacy interpolation syntax.

Hope that helps!

Ryan